.vib/mongodb-sharded/ginkgo/mongodb_test.go

```diff
@@ -0,0 +1,223 @@
+				*releaseName, jobSuffix)
+			err = createJob(ctx, c, deleteDBJobName, port, image, fmt.Sprintf("db.%s.drop()", dbName))
+			Expect(err).NotTo(HaveOccurred())
+
+			Eventually(func() (*batchv1.Job, error) {
+				return c.BatchV1().Jobs(*namespace).Get(ctx, deleteDBJobName, metav1.GetOptions{})
+			}, Timeout, PollingInterval).Should(WithTransform(getSucceededPods, Equal(int32(1))))
+		})
+	})
+
+	AfterEach(func() {
+		cancel()
+	})
+})
+
+func clusterConfigOrDie() *rest.Config {
+	if *kubeconfig == "" {
+		panic("kubeconfig must be supplied")
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return config
+}
+
+func getAvailableReplicas(ss *appsv1.StatefulSet) int32 {
+	return ss.Status.AvailableReplicas
+}
+
+func getSucceededPods(j *batchv1.Job) int32 {
+	return j.Status.Succeeded
+}
+
+func createJob(ctx context.Context, c kubernetes.Interface, name, port, image, stmt string) error {
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Job",
+		},
+		Spec: batchv1.JobSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					RestartPolicy: "Never",
+					Containers: []v1.Container{
+						{
+							Name:    "mongodb",
+							Image:   image,
+							Command: []string{"mongosh", "--quiet", "--username", *username, "--password", *password, "--host", *releaseName, "--port", port, "--eval", stmt},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, err := c.BatchV1().Jobs(*namespace).Create(ctx, job, metav1.CreateOptions{})
+
+	return err
+}
+
+func scale(ctx context.Context, c kubernetes.Interface, ss *appsv1.StatefulSet, count int32) (*appsv1.StatefulSet, error) {
+	name := ss.Name
+	ns := ss.Namespace
+
+	for i := 0; i < 3; i++ {
+		ss, err := c.AppsV1().StatefulSets(ns).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get deployment %q: %v", name, err)
+		}
+		*(ss.Spec.Replicas) = count
+		ss, err = c.AppsV1().StatefulSets(ns).Update(ctx, ss, metav1.UpdateOptions{})
+		if err == nil {
+			return ss, nil
+		}
+		if !apierrors.IsConflict(err) && !apierrors.IsServerTimeout(err) {
+			return nil, fmt.Errorf("failed to update statefulset %q: %v", name, err)
+		}
+	}
+
+	return nil, fmt.Errorf("too many retries draining statefulset %q", name)
+}
+
+func getContainerImage(ss *appsv1.StatefulSet, name string) (string, error) {
+	containers := ss.Spec.Template.Spec.Containers
+
+	for _, c := range containers {
+		if c.Name == name {
+			return c.Image, nil
+		}
+	}
+
+	return "", fmt.Errorf("container %q not found in statefulset %q", name, ss.Name)
+}
+
+func getPort(svc *v1.Service, name string) (string, error) {
+	ports := svc.Spec.Ports
+
+	for _, p := range ports {
+		if p.Name == name {
+			return strconv.Itoa(int(p.Port)), nil
+		}
+	}
+
+	return "", fmt.Errorf("port %q not found in service %q", name, svc.Name)
+}
```

Comment:
All these functions should be part of a common library shared between tests, right?